### PR TITLE
Bugfix hook backend detection

### PR DESCRIPTION
**Pull request by BenjamenMeyer**
- Bug Fix: Configuration File did not really have a 'disk' configuration. Updated it to do so.
- Bug Fix: The Hooks seemed to be loaded incorrectly; adjusted usage of loading the hooks through falcon.API() to be more inline with the Falcon documentation; Falcon now takes care of calling the hooks for us
- Enhancement: Updated the driver configurations so that the initial driver spec simply specifies a sub-section of the configuration for which to load the driver
  - f.e deuce.conf.metadata_driver.driver now is either 'sqlite', 'cassandra', or 'mongodb', the actual driver is specified in that section's driver, f.e deuce.conf.metadata_driver.sqlite.driver
  - raises an exception specific to the driver type if the specified driver for metadata or storage is not recognized
- Enhancement: Uses the deuce.conf.block_storage_driver.driver to determine whether to load the Swift Hooks
  - extensible so we could easily add other values as well
- Bug Fix: Updated the configspec.ini to map more of the values from the specific sections
  - these will typically be provided by default if the user copies the example config.ini and most won't need to be changed
- Enhancement: Updated configspec to use specific values for the drivers
- Enhancement: Updated configspec so the cassandra cluster address space is validated as a list of strings

### CODE REVIEW

**Comment by racklanta** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://ci.cldbckp.com/job/deuce-pullrequest/257/

File: deuce/tests/test_model_init.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import unittest
```
Comment (TheSriram) [inline deuce/tests/test_model_init.py]: un-needed import?

File: ini/config.ini [hunk 2]
```diff
@@ -43,8 +43,9 @@ log_directory = log
         format = '%(asctime)s %(levelname)-5.5s [%(name)s/%(lineno)d][%(threadName)s] [%(request_id)s] : %(message)s'
 
 [block_storage_driver]
-driver = deuce.drivers.disk.DiskStorageDriver
-    [[options]]
+driver = disk
+    [[disk]]
+		driver = deuce.drivers.disk.DiskStorageDriver
```
Comment (TheSriram) [inline ini/config.ini]: Seems to be offset by a tab

File: ini/config.ini [hunk 3]
```diff
@@ -57,8 +58,9 @@ driver = deuce.drivers.disk.DiskStorageDriver
             storage_url = Storage Url
 
 [metadata_driver]
-driver = deuce.drivers.sqlite.SqliteStorageDriver
+driver = sqlite
     [[cassandra]]
+		driver = deuce.drivers.cassandra.CassandraStorageDriver
```
Comment (TheSriram) [inline ini/config.ini]: same here

File: ini/configspec.ini [hunk 4]
```diff
@@ -9,17 +9,36 @@ datacenter = string
...
+	url = string
+	db_file = string
+	db_module = string
     FileBlockReadSegNum = integer
     maxFileBlockSegNum = integer
         [[[testing]]]
         is_mocking = boolean
     [[cassandra]]
+    driver = string
+	cluster = string_list
```
Comment (TheSriram) [inline ini/configspec.ini]: I think setting a ',' at the end in the config.ini automatically makes it a list. But being explicit helps.

File: ini/configspec.ini [hunk 5]
```diff
@@ -9,17 +9,36 @@ datacenter = string
 default_returned_num = integer
 max_returned_num = integer
 [metadata_driver]
+driver = option('sqlite', 'mongodb', 'cassandra')
+    [[sqlite]]
+    driver = string
```
Comment (TheSriram) [inline ini/configspec.ini]: Arent they strings by default?

File: ini/configspec.ini (hunk 4 above)
Comment (BenjamenMeyer) [inline ini/configspec.ini]: There wasn't anything there prior, and it's keeps getting mentioned (in passing) to specify all the servers for failover so I figured it'd be better to be explicit.

However, this also means that if there is only a single entry that it must end in a comma (,).

File: ini/configspec.ini (hunk 5 above)
Comment (BenjamenMeyer) [inline ini/configspec.ini]: Not aware but I wanted to ensure these entries existed.

**Comment by TheSriram** [comment]
:+1:

**Comment by racklanta** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://ci.cldbckp.com/job/deuce-pullrequest/259/

**Comment by BenjamenMeyer** [comment]
@dstritzel FYI - this PR will have a change on the configuration scripts. It should actually make it easier/nicer for the engineering team.

**Comment by painterjd** [comment]
This is base on #247 which is not being merged as-is, so :-1:

**Pull request closed without merging**